File: SESION3/C++/FiltroJugadores.cpp
using namespace std;

#include "FiltroJugadores.h"
#include "Alquiler.h"
#include "Ubicacion.h"
#include "Filtro.h"
#include <string>

void FiltroJugadores::aplicarFiltro(Alquiler & alq) {

    
    string loc= alq.getLocalizacion();

    for( auto item : ubicaciones){
        if(item.getLocalizacion() == loc){
            alq.incrementarCoste(item.getPrecioJugador()*alq.getNumeroJugadores()); //Precio por numero de jugadores
        }
    }


}

File: P2/practica2/lib/filtro-jugadores.dart
import 'package:practica2/ubicacion.dart';
import 'package:practica2/filtro.dart';
import 'package:practica2/alquiler.dart';


class FiltroJugadores extends Filtro
{

        FiltroJugadores(){}
        
        void aplicarFiltro(Alquiler alq){

            String loc = alq.getLocalizacion();

            for (final x in ubicaciones){
              if (x.getLocalizacion() == loc)
                alq.incrementarCoste(x.getPrecioJugador()*alq.getNumeroJugadores());
            }

        }
}
